dsl_parser/blueprint_diff.py

```diff
@@ -0,0 +1,121 @@
+########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+
+def blueprint_diff(old_blueprint, new_blueprint):
+
+    chain = []
+    output = {}
+
+    diff(old_blueprint, new_blueprint, chain, output, 'added')
+    diff(new_blueprint, old_blueprint, chain, output, 'removed',
+         include_modified=False)
+
+    return output
+
+
+def diff(old, new, chain, output, mode, include_modified=True):
```

Comment:
change chain to 'path' or something similar

that also reflects on the `key` var in the main loop